lib/Conversion/TritonGPUToLLVM/Utility.cpp

```diff
@@ -275,7 +278,13 @@ Value storeShared(ConversionPatternRewriter &rewriter, Location loc, Value ptr,
 Value loadShared(ConversionPatternRewriter &rewriter, Location loc, Value ptr,
                  Value pred) {
 #if USE_ROCM
-  return load(ptr);
+  auto loaded = rewriter.create<scf::IfOp>(loc, pred,
+    [&](OpBuilder& builder, Location loc) {
+      load(ptr);
+    },
+    nullptr
+  );
```

Comment:
There is no return value from if operation, so `getResult(0)` will break,
try something like this:
```suggestion
  auto loaded = rewriter.create<scf::IfOp>(loc, pred,
    [&](OpBuilder& builder, Location loc) {
      auto loadVal = load(ptr);
      builder.create<mlir::scf::YieldOp>(loc, ValueRange({loadVal}));
    },
    [&](OpBuilder& builder, Location loc) {
      Value zeroConst;
      // for fp16
      zeroConst = f16_val(0.0);
      builder.create<mlir::scf::YieldOp>(loc, ValueRange({zeroConst}));
    }
  );
```